js/src/components/shipping-rate-section/estimated-shipping-rates-card/getHandlers.test.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * Internal dependencies
+ */
+import getHandlers from './getHandlers';
+
+describe( 'getHandlers', () => {
+	let mockOnChange;
+	let handlers;
+
+	beforeEach( () => {
+		const value = [
+			{
+				id: '1',
+				country: 'US',
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 20,
+				options: {},
+			},
+			{
+				id: '2',
+				country: 'AU',
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 20,
+				options: {
+					free_shipping_threshold: 50,
+				},
+			},
+			{
+				id: '3',
+				country: 'CN',
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 25,
+				options: {
+					free_shipping_threshold: 50,
+				},
+			},
+		];
+
+		mockOnChange = jest.fn();
+		handlers = getHandlers( {
+			value,
+			onChange: mockOnChange,
+		} );
+	} );
+
+	describe( 'handleAddSubmit', () => {
+		it( 'returns value with the newly added group', () => {
+			const newGroup = {
+				countries: [ 'MY', 'SG' ],
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 30,
+			};
+
+			const { handleAddSubmit } = handlers;
+			handleAddSubmit( newGroup );
+
+			expect( mockOnChange.mock.calls.length ).toBe( 1 );
+			expect( mockOnChange.mock.calls[ 0 ][ 0 ] ).toStrictEqual( [
+				{
+					id: '1',
+					country: 'US',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {},
+				},
+				{
+					id: '2',
+					country: 'AU',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+				{
+					id: '3',
+					country: 'CN',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 25,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+				{
+					country: 'MY',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 30,
+					options: {},
+				},
+				{
+					country: 'SG',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 30,
+					options: {},
+				},
+			] );
+		} );
+	} );
+
+	describe( 'getChangeHandler', () => {
+		it( 'returns value updated based on changed group rate', () => {
```

Comment:
:construction: (here and below)
"When called with a value, returns a function, which when called with a new value, calls `onChange` callback with a changed...."

:nail_care: :grey_question: 
Just writing the above gives me a smell that maybe this state keeping logic may be a tad overcomplicated